src/Creative/InLine/Linear.php

```diff
@@ -60,6 +66,23 @@ public function createMediaFile()
         return new MediaFile($mediaFileDomElement);
     }
 
+    /**
+     * @return AdParameters
+     */
+    public function createAdParameters()
+    {
+        if (empty($this->adParametersDomElement)) {
```

Comment:
i thing you copy&paste this code from createMediaFile. There mediaFiles is a collection of mediaFile elements, and `if (empty($this->mediaFilesDomElement)) {` used to lazy create container. Here you got same node when will call createAdParameters second time.

Reply:
proposal applied, make it production ready )